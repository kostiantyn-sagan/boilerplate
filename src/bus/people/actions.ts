// API;
import { api } from './api';

// Types;
import { types } from './types';

export const peopleActions = Object.freeze({
    // Sync
    startFetching: () => {
        return {
            type: types.PEOPLE_START_FETCHING,
        };
    },
    stopFetching: () => {
        return {
            type: types.PEOPLE_STOP_FETCHING,
        };
    },
    fill: (payload:any) => {
        return {
            type: types.PEOPLE_FILL,
            payload,
        };
    },
    setFetchingError: (error:any) => {
        return {
            type:    types.PEOPLE_SET_FETCHING_ERROR,
            error:   true,
            payload: error,
        };
    },

    fetchAsync: (id?:string) => async(dispatch:any) => {
        dispatch(peopleActions.startFetching());

        const response = await api.people.fetch(id);

        if (response.status === 200) {
            const  results  = await response.json();

            const data = id ? results : results.results;


            dispatch(peopleActions.fill(data));
        } else {
            const error = {
                status: response.status,
            };

            dispatch(peopleActions.setFetchingError(error));
        }
        dispatch(peopleActions.stopFetching());
    },
});
