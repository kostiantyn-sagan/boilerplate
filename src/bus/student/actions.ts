import { types } from './types';

export const studentActions = {
    fillProfile: (profile:any) => {
        return {
            type:    types.STUDENT_FILL,
            payload: profile,
        };
    },
};
