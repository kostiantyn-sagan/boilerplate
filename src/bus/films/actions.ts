// API;
import { api } from '../people/api';

// Types;
import { types } from './types';

export const filmsActions = Object.freeze({
    // Sync
    startFetching: () => {
        return {
            type: types.FILMS_START_FETCHING,
        };
    },
    stopFetching: () => {
        return {
            type: types.FILMS_STOP_FETCHING,
        };
    },
    fill: (payload:any) => {
        return {
            type: types.FILMS_FILL,
            payload,
        };
    },
    setFetchingError: (error:any) => {
        return {
            type:    types.FILMS_SET_FETCHING_ERROR,
            error:   true,
            payload: error,
        };
    },

    fetchAsync: (id?:string) => async(dispatch:any) => {
        dispatch(filmsActions.startFetching());

        const response = await api.films.fetch(id);

        if (response.status === 200) {
            const  results  = await response.json();
            const data = id ? results : results.results;

            dispatch(filmsActions.fill(data));
        } else {
            const error = {
                status: response.status,
            };

            dispatch(filmsActions.setFetchingError(error));
        }
        dispatch(filmsActions.stopFetching());
    },
});
