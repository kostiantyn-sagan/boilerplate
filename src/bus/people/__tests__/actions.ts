import { peopleActions } from '../actions';

describe('people actions', () => {
    test('should start fetching', () => {
        expect(peopleActions.startFetching()).toMatchSnapshot();
    });
    test('should stop fetching', () => {
        expect(peopleActions.stopFetching()).toMatchSnapshot();
    });
    test('should fetch people', () => {
        expect(peopleActions.fetchAsync()).toMatchSnapshot();
    });
    test('should fill people', () => {
        expect(peopleActions.fill({ name: 'Mars' })).toMatchSnapshot();
    });
    test('should set fetching error', () => {
        expect(peopleActions.setFetchingError('We have an error!')).toMatchSnapshot();
    });
});
