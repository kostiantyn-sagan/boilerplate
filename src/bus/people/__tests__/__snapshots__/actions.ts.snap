// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`people actions should fetch people 1`] = `[Function]`;

exports[`people actions should fill people 1`] = `
Object {
  "payload": Object {
    "name": "Mars",
  },
  "type": "PEOPLE_FILL",
}
`;

exports[`people actions should set fetching error 1`] = `
Object {
  "error": true,
  "payload": "We have an error!",
  "type": "PEOPLE_SET_FETCHING_ERROR",
}
`;

exports[`people actions should start fetching 1`] = `
Object {
  "type": "PEOPLE_START_FETCHING",
}
`;

exports[`people actions should stop fetching 1`] = `
Object {
  "type": "PEOPLE_STOP_FETCHING",
}
`;
